### PR TITLE
thespian: Python library init at 3.8.3

### PR DESCRIPTION
**Pull request by kquick**
###### Motivation for this change
New package (library for Python).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/thespian/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    extension = "zip";
+    sha256 = "0vvwsh3waxd5ldrayr86kdcshv07bp361fl7p16g9i044vklwly4";
+    # fdbbea4edb6bad0ac0e53fc7bc6970e78e12eef4944aa4146bcdcb573201676c";
+  };
+
+  # Do not run the test suite: it takes a long type and uses
+  # significant system resources.  Thespian tests are performed via
+  # it's Travis CI configuration and do not need to be duplicated
+  # here.
+  doCheck = false;
```
Comment (Mic92) [inline pkgs/development/python-modules/thespian/default.nix]: The problem is here less the time spent, but that it requires network for most tests.

File: pkgs/development/python-modules/thespian/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+buildPythonPackage rec {
+  version = "3.8.3";
+  pname = "thespian";
+  name = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    extension = "zip";
+    sha256 = "0vvwsh3waxd5ldrayr86kdcshv07bp361fl7p16g9i044vklwly4";
+    # fdbbea4edb6bad0ac0e53fc7bc6970e78e12eef4944aa4146bcdcb573201676c";
```
Comment (FRidh) [inline pkgs/development/python-modules/thespian/default.nix]: can you remove this

File: pkgs/development/python-modules/thespian/default.nix (hunk 1 above)
Comment (FRidh) [reply pkgs/development/python-modules/thespian/default.nix]: Indeed, please update the comment.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by kquick** [comment]
@FRidh, @Mic92 thanks for catching that, I've pushed updates to correct that.

**Comment by FRidh** [comment]
> Thespian tests are performed via it's Travis CI configuration and do not need to be duplicated here.

Why not? Do you mean with that, that we don't have to test a package if upstream tests it already?

**Comment by kquick** [comment]
@FRidh the thespian tests are oriented toward functional testing to verify that the process management and network communications provided by the package are working correctly.  During the testing it will create hundreds of processes and use a large number of network ports and corresponding resources.  The testing typically takes on the order of 25 minutes to complete.   They certainly could be enabled, but it was my thought that this would be either a bad user experience or a high burden on the hydra system already under stress.  The comment in question was not intended to absolve of the need for general testing, but rather intended to reassure the viewer that there *was* testing performed even though it wasn't performed immediately upon installation.

Do you have a recommendation for a better approach or better wording?

**Comment by FRidh** [comment]
Ideally a limited amount of testing is done to determine the package is functioning. The absolute minimum would be checking imports, although typically we do not take the effort to implement this, and just have tests disabled.

**Pull request merged**